XamlControlsGallery/Navigation/NavigationRootPage.xaml.cs

```diff
@@ -211,6 +214,9 @@ private void OnNavigationViewItemInvoked(Microsoft.UI.Xaml.Controls.NavigationVi
 
         private void OnRootFrameNavigated(object sender, NavigationEventArgs e)
         {
+            // Close any open teaching tips before navigation
+            CloseTeachingTips();
+
```

Comment:
Is this necessary because the tips are not light dismiss? I wonder if this points at a programmability problem with TeachingTip. Should targeted tips go away when their target is unloaded?

Reply:
Also this should be in the TeachingTipPage itself, I would think. Like in OnNavigatingAway?

---
In reply to: [268937523](https://github.com/Microsoft/Xaml-Controls-Gallery/pull/73#discussion_r268937523) [](ancestors = 268937523)

Reply:
+1 to the concern that the app dev needs to do work to get rid of Tips during navigation. I think targeted tips going away when unloaded makes a ton of sense.

In the meantime, I think listening to OnNavigatingAway would probably work and result in a more elegant (and encapsulated) experience.

Reply:
For historical context, I believe the reason we decided against facilitating the closing of persistent tips is because we also do not provide any assistance in opening opening one and, at the time, there was all or nothing mindset on adding a show/close service that was ultimately adopted by the Context Experience team for 1st party apps. 

Now that I am seeing this in action, I am also +1 on the value of closing persistent tips when their target is unloaded.

Reply:
Please file an issue in the MUX github. It feels like something we should do.

Reply:
Will do. Would you suggest labeling it as a bug or a feature request?

Reply:
You can just open a regular issue and we'll label it in triage.